File: network-equipment-pms/pms-system/src/main/java/com/dp/plat/system/entity/Feedback.java
package com.dp.plat.system.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.dp.plat.common.entity.BaseEntity;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * User feedback / support ticket entity.
 *
 * <p>Stores bug reports, suggestions, questions and other feedback submitted by
 * authenticated users. Administrators can reply to or close the ticket; the
 * lifecycle is tracked via {@link #status} (PENDING → PROCESSING → RESOLVED →
 * CLOSED).</p>
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
@TableName("sys_feedback")
public class Feedback extends BaseEntity {

    /** 提交人用户 ID（冗余字段，便于按用户筛选） */
    private Long userId;

    /** 提交人用户名（冗余字段，便于展示） */
    @Size(max = 50, message = "用户名长度不能超过 50 个字符")
    private String username;

    /** 反馈分类：BUG / SUGGESTION / QUESTION / OTHER */
    @NotBlank(message = "分类不能为空")
    @Pattern(regexp = "^(BUG|SUGGESTION|QUESTION|OTHER)$",
            message = "分类只能为 BUG / SUGGESTION / QUESTION / OTHER")
    private String category;

    @NotBlank(message = "标题不能为空")
    @Size(max = 200, message = "标题长度不能超过 200 个字符")
    private String title;

    @NotBlank(message = "内容不能为空")
    @Size(max = 4000, message = "内容长度不能超过 4000 个字符")
    private String content;

    /** 联系方式（选填，电话 / 邮箱） */
    @Size(max = 100, message = "联系方式长度不能超过 100 个字符")
    private String contact;

    /** 状态：PENDING / PROCESSING / RESOLVED / CLOSED */
    @Pattern(regexp = "^(PENDING|PROCESSING|RESOLVED|CLOSED)$",
            message = "状态只能为 PENDING / PROCESSING / RESOLVED / CLOSED")
    private String status;

    /** 管理员回复内容 */
    @Size(max = 4000, message = "回复内容长度不能超过 4000 个字符")
    private String reply;

    /** 回复人用户名 */
    @Size(max = 50, message = "回复人长度不能超过 50 个字符")
    private String replyBy;

    /** 回复时间 */
    private LocalDateTime replyAt;
}

File: network-equipment-pms/pms-frontend/src/api/feedback.ts
import { get, post, put } from '@/utils/request'

/** 反馈分类 */
export type FeedbackCategory = 'BUG' | 'SUGGESTION' | 'QUESTION' | 'OTHER'

/** 反馈状态 */
export type FeedbackStatus = 'PENDING' | 'PROCESSING' | 'RESOLVED' | 'CLOSED'

/** 反馈实体（与后端 com.dp.plat.system.entity.Feedback 对应） */
export interface Feedback {
  id?: number
  userId?: number
  username?: string
  category: FeedbackCategory
  title: string
  content: string
  /** 联系方式（电话/邮箱，选填） */
  contact?: string
  status?: FeedbackStatus
  reply?: string
  replyBy?: string
  replyAt?: string
  createTime?: string
  updateTime?: string
}

/** 反馈回复请求体 */
export interface FeedbackReplyRequest {
  reply: string
}

/** 按状态统计的反馈数量 */
export type FeedbackStatusStats = Record<FeedbackStatus, number>

/** 提交新反馈（任意已登录用户） */
export function createFeedback(data: Feedback): Promise<boolean> {
  return post<boolean>('/api/system/feedback', data)
}

/** 获取反馈详情（创建者本人或管理员） */
export function getFeedback(id: number): Promise<Feedback> {
  return get<Feedback>(`/api/system/feedback/${id}`)
}

/** 列出当前用户提交的反馈 */
export function listMyFeedbacks(): Promise<Feedback[]> {
  return get<Feedback[]>('/api/system/feedback/my')
}

/** 列出所有反馈（管理员） */
export function listAllFeedbacks(
  status?: FeedbackStatus,
  category?: FeedbackCategory
): Promise<Feedback[]> {
  return get<Feedback[]>('/api/system/feedback/list', {
    ...(status ? { status } : {}),
    ...(category ? { category } : {})
  })
}

/** 回复反馈（管理员） */
export function replyFeedback(id: number, data: FeedbackReplyRequest): Promise<boolean> {
  return put<boolean>(`/api/system/feedback/${id}/reply`, data)
}

/** 关闭反馈（管理员） */
export function closeFeedback(id: number): Promise<boolean> {
  return put<boolean>(`/api/system/feedback/${id}/close`)
}

/** 按状态统计当前用户的反馈数量 */
export function getFeedbackStatusStats(): Promise<FeedbackStatusStats> {
  return get<FeedbackStatusStats>('/api/system/feedback/stats')
}
